pkg/common-controller/snapshot_controller.go

```diff
@@ -190,6 +193,21 @@ func (ctrl *csiSnapshotCommonController) syncSnapshot(snapshot *crdv1.VolumeSnap
 	return nil
 }
 
+// Check if PVC is being used by the current snapshot as source
+func (ctrl *csiSnapshotCommonController) isPVCInUseByCurrentSnapshot(snapshot *crdv1.VolumeSnapshot) bool {
+	// Get snapshot source which is a PVC
+	pvc, err := ctrl.getClaimFromVolumeSnapshot(snapshot)
```

Comment:
IMO, it might worth checking whether the PVC has the finalizer here as well. It does not make sense to block the deletion if there is no finalizer on PVC?

Reply:
Done